### PR TITLE
Switch CI to GitHub Actions.

### PR DESCRIPTION
**Pull request by clokep**
This is cribbed from https://hynek.me/articles/python-github-actions/ + my own projects.

### CODE REVIEW

**Comment by clokep** [comment]
Seems that tox isn't running in the directory that I expect it to here? Not sure what's going on there.

**Comment by anoadragon453** [comment]
>  py run-test: commands[1] | coverage run --include=s3_storage_provider.py -m trial test_s3
> No module named 'trial'

Is that supposed to be `twisted.trial`?

**Comment by clokep** [comment]
🤦

File: tox.ini [hunk 1]
```diff
@@ -19,12 +19,8 @@ passenv = *
 
 commands =
     /usr/bin/find "{toxinidir}" -name '*.pyc' -delete
-    coverage run {env:COVERAGE_OPTS:} --include=s3_storage_provider.py \
-        "{envbindir}/trial" {env:TRIAL_FLAGS:} {posargs:test_s3} {env:TOXSUFFIX:}
-    {env:DUMP_COVERAGE_COMMAND:coverage report -m}
```
Comment (anoadragon453) [inline tox.ini:24]: Given we're dropping these environment variables, can we also remove the `PEP8SUFFIX` here? https://github.com/matrix-org/synapse-s3-storage-provider/blob/236e0cddb8e176debccd77321a5d6ed6d44dc617/tox.ini#L53

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+      - name: "Install dependencies"
+        run: |
+          set -xe
+          python -VV
+          python -m site
+          python -m pip install --upgrade pip setuptools wheel
+          python -m pip install --upgrade tox
+      - name: "Run tox lint targets for ${{ matrix.python-version }}"
+        run: "python -m tox -e check_isort,pep8,packaging"
```
Comment (anoadragon453) [inline .github/workflows/ci.yml:31]: How come we do `python -m tox` here and just `tox` below?

**Review by anoadragon453 — COMMENTED** [review]
Just a couple small things, but overall looks good!

File: .github/workflows/ci.yml (hunk 2 above)
Comment (clokep) [reply .github/workflows/ci.yml:31]: I don't think there's a particular reason. Probably because I copied these lines out of different configs.

**Review by anoadragon453 — APPROVED** [review]

**Pull request merged**